GTG/gtk/preferences.py

```diff
@@ -17,304 +17,50 @@
-                                'on_pref_autostart_toggled':
-                                self.on_autostart_toggled,
-                                'on_pref_show_preview_toggled':
-                                self.toggle_preview,
-                                'on_bg_color_toggled':
-                                self.on_bg_color_toggled,
-                                'on_prefs_help':
-                                self.on_help,
-                                'on_prefs_close':
-                                self.on_close,
-                                'on_PreferencesDialog_delete_event':
-                                self.on_close,
-                                'on_fontbutton_font_set':
-                                self.on_font_change,
-                                'on_shortcut_button_toggled':
-                                self.shortcut.on_shortcut_toggled,
-                                'on_valid_check':
-                                self.valid_check,
-                                })
+        self.headerbar = builder.get_object("right_header_bar")
+        self.stack = builder.get_object("stack")
 
-    def _refresh_preferences_store(self):
-        """ Sets the correct value in the preferences checkboxes """
-        has_autostart = os.path.isfile(AUTOSTART_PATH)
-        self.pref_autostart.set_active(has_autostart)
+        self.general_pref = GeneralPreferences(req, vmanager)
+        general = self.general_pref.get_ui()
+        self.stack.add_titled(general, "general", "General Preferences")
 
-        self.shortcut.refresh_accel()
-
-        show_preview = self.config.get("contents_preview_enable")
-        self.pref_show_preview.set_active(show_preview)
-
-        bg_color = self.config.get("bg_color_enable")
-        self.bg_color_enable.set_active(bg_color)
-        refresh_hour, refresh_mins = self.timer.get_configuration()
-        self.refresh_hour.set_text(refresh_hour)
-        self.refresh_mins.set_text(refresh_mins)
-
-    def _refresh_task_browser(self):
-        """ Refresh tasks in task browser """
-        task_tree = self.req.get_tasks_tree(refresh=False).get_basetree()
-        task_tree.refresh_all()
+        self.on_sidebar_change(self.stack)
 
     def activate(self):
-        """ Activate the preferences dialog."""
-        self._refresh_preferences_store()
-        self.dialog.show_all()
-
-    def valid_check(self, widget):
-        try:
-            self.timer.set_configuration(self.refresh_hour.get_text(),
-                                         self.refresh_mins.get_text())
-            color = None
-        except (ValueError, TypeError):
-            color = self.color_invalid
-
-        self.refresh_hour.modify_fg(Gtk.StateFlags.NORMAL, color)
-        self.refresh_mins.modify_fg(Gtk.StateFlags.NORMAL, color)
+        """ Activate the preferences window."""
+        self.general_pref._refresh_preferences_store()
+        self.window.show_all()
```

Comment:
You can probably easily fix .ui files to make sure they have visibility=True set